### PR TITLE
UML-261 Added filter to dob input field and added additional tests

### PR DESCRIPTION
**Pull request by SeemaMenon**

### CODE REVIEW

File: service-front/app/src/Actor/src/Form/LpaAdd.php [hunk 1]
```diff
@@ -103,6 +104,9 @@ public function getInputFilterSpecification() : array
                 ]
             ],
             'dob' => [
+                'filters'  => [
+                    ['name' => DatePrefixFilter::class],
```
Comment (cooperaj) [inline service-front/app/src/Actor/src/Form/LpaAdd.php]: Please ensure consistency of formatting. Either split the array over multiple lines as with the validator below or change the validator to match this one.

File: service-front/app/src/Actor/src/Handler/LpaAddHandler.php [hunk 2]
```diff
@@ -73,11 +73,13 @@ public function handle(ServerRequestInterface $request) : ResponseInterface
 
                 //  Convert the date of birth
                 $dobString = sprintf('%s-%s-%s', $postData['dob']['year'], $postData['dob']['month'], $postData['dob']['day']);
+                //$dobFormatted = sprintf("%04d-%02d-%02d", $postData['dob']['year'], $postData['dob']['month'], $postData['dob']['day']);
```
Comment (cooperaj) [inline service-front/app/src/Actor/src/Handler/LpaAddHandler.php]: Please don't commit commented lines.

File: service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php]: Please remove the PHPStorm added comment. 

Probs worth editing your new file templates to setup correctly. You should also be declaring the file with strict types.

```
<?php

declare(strict_types=1);

namespace ...
```

File: service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * Created by PhpStorm.
+ * User: seemamenon
+ * Date: 11/11/2019
+ * Time: 14:57
+ */
+
+namespace Common\Form\Fieldset;
+
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php]: Extra line

File: service-front/app/test/CommonTest/Form/Fieldset/DobPrefixFilterTest.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                    'year' => '1980',
+                ]
+            ],
+        ];
+    }
+}
+
+
+
+
```
Comment (cooperaj) [inline service-front/app/test/CommonTest/Form/Fieldset/DobPrefixFilterTest.php]: Seems to be a lot of extra lines here. Odd as PHPStorm should auto-truncate files to 1 newline only.

File: service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * Defined by Zend\Filter\FilterInterface
+     *
+     * Returns the string $value with zero prefixed at the beginning of the value
+     *
+     * @param  array $value
+     * @return string
+     */
+    public function filter($value)
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php]: with strict types you should be setting the parameter and return types on the function signature

`public function filter(array $value): string`

File: service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        return $value;
+    }
+
+    /**
+     *
+     * @param array $value
+     * @param string $formattedDate
+     * @return string
+     */
+    protected function formatWithLeadingZero($value)
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php]: `protected function formatWithLeadingZero(string $value): string`

File: service-front/app/src/Actor/src/Form/LpaAdd.php (hunk 1 above)
Comment (SeemaMenon) [reply service-front/app/src/Actor/src/Form/LpaAdd.php]: done

File: service-front/app/src/Actor/src/Handler/LpaAddHandler.php (hunk 2 above)
Comment (SeemaMenon) [reply service-front/app/src/Actor/src/Handler/LpaAddHandler.php]: removed

File: service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php (hunk 3 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php]: done

File: service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php (hunk 4 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php]: removed

File: service-front/app/test/CommonTest/Form/Fieldset/DobPrefixFilterTest.php (hunk 5 above)
Comment (SeemaMenon) [reply service-front/app/test/CommonTest/Form/Fieldset/DobPrefixFilterTest.php]: removed

File: service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php (hunk 6 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php]: Done..provided return type  to be array

File: service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php (hunk 7 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Form/Fieldset/DatePrefixFilter.php]: Done

**Pull request merged**